widget/markdown_test.go

```diff
@@ -163,29 +163,29 @@ func TestRichTextMarkdown_Image(t *testing.T) {
 func TestRichTextMarkdown_Lines(t *testing.T) {
 	r := NewRichTextFromMarkdown("line1\nline2\n") // a single newline is not a new paragraph
 
-	assert.Equal(t, 2, len(r.Segments))
+	assert.Equal(t, 3, len(r.Segments))
 	if text, ok := r.Segments[0].(*TextSegment); ok {
-		assert.Equal(t, "line1 ", text.Text)
+		assert.Equal(t, "line1", text.Text)
 		assert.True(t, text.Inline())
 	} else {
 		t.Error("Segment should be Text")
 	}
 	if text, ok := r.Segments[1].(*TextSegment); ok {
-		assert.Equal(t, "line2", text.Text)
+		assert.Equal(t, " line2", text.Text)
```

Comment:
Just coming back to this apologies for the delay. Is this a desirable change? Putting a space at the leading of a string seems like it could cause a problem.

Reply:
Previously trailing spaces were inserted, now leading spaces are inserted instead. I cannot remember why I changed it, but I guess it made the implementation a little easier. I don't see how either choice would be better than the other. What kind of problem do you foresee?

Reply:
The problem I foresee is that strings are indexed from the beginning, so if anyone was actually working with the content of a TextSegment then the characters inside them are now off-by-1.

Reply:
Good point. I'll try to revert back to trailing spaces when I find some time.

Reply:
I have now revisited the code and was able to return to trailing spaces instead of leading ones. In the process, the code even became a little more compact and the `renderNode` function requires one less argument.

I hope I didn't forget anything that I had considered when originally creating this pull request, but the tests cannot find any issues.